client/src/components/common/Navbar.js

```diff
@@ -1,46 +1,161 @@
+    height: '100vh',
+    display: 'flex',
+    flexFlow: 'column nowrap',
+    justifyContent: 'space-around',
+    alignItems: 'stretch',
+    textAlign: 'center'
+  },
+  text: {
+    textAlign: 'center'
+  },
+  logo: {
+    width: '64px',
+    height: '64px',
+    '&:hover': {
+      cursor: 'pointer'
+    }
+  }
+})
+
 
 const Navbar = props => {
-  const { classes, to, ...other } = props;
-  const primary = breadcrumbNameMap[to];
+  const { classes, theme } = props;
+  console.log('THEME', theme)
+  const [open, setOpen] = useState(false)
+
+  const toggleDrawer = () => {
+    setOpen(!open)
+  }
+
+  const drawer = (
+    <List component="nav" className={classes.list} >
+      <ListItemLink to="/dashboard" className={classes.text} />
+      <ListItemLink to="/refreshrs" className={classes.text} />
+      <ListItemLink to="/classes" className={classes.text} />
+      <ListItemLink to="/billing" className={classes.text} />
+    </List>
+  )
 
   return (
-    <AppBar position="static">
-    
-    <Toolbar className={classes.nav}>
-   
-      <Tabs >
-        <Tab to="/dashboard" primary={primary} component={RouterLink} {...other} label='dashboard'/>
-        <Tab to="/refreshrs" primary={primary} component={RouterLink} {...other} label='refreshrs'/>
-        <Tab to="/classes" primary={primary} component={RouterLink} {...other} label='classes'/>
-        <Tab to="/billing" primary={primary} component={RouterLink} {...other} label='billing'/>
-      </Tabs>
-
-      <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
+    <div className={classes.root}>
+      <AppBar position="fixed" className={classes.appBar}>
+        <Toolbar className={classes.toolbar}>
+          <Typography variant="h6" color="inherit" noWrap>
+            Logout
+          </Typography>
+          <HomeLink className={classes.logo} to='/home' />
+          <IconButton color="inherit" aria-label="Open Nav" onClick={toggleDrawer} className={classes.menuButton}>
             <MenuIcon />
           </IconButton>
-    </Toolbar>
-    </AppBar>
+        </Toolbar>
+      </AppBar>
+      <nav className={classes.drawer}>
+        <Hidden smUp implementation="js">
+          <Drawer
+            variant="temporary"
+            anchor='right'
+            open={open}
+            onClose={toggleDrawer}
+            classes={{
+              paper: classes.drawerPaper,
+            }}
+          >
+            {drawer}
+          </Drawer>
+        </Hidden>
+        <Hidden xsDown implementation="css">
+          <Drawer
+            anchor='right'
+
+            classes={{
+              paper: classes.drawerPaper,
+            }}
+            variant="permanent"
+            open
+          >
+            {drawer}
+          </Drawer>
+        </Hidden>
+      </nav>
+    </div>
   );
 };
 
-export default withRouter(withStyles(styles)(Navbar));
+export default withRouter(withStyles(styles, { withTheme: true })(Navbar));
```

Comment:
What does `withTheme: true` do?

Reply:
It allows you to use the theme object within the style object, otherwise it will give an undefined error for theme when doing theme.spacing.unit for example